etcdctl/ctlv3/command/ep_command.go

```diff
@@ -124,14 +125,32 @@ func epHealthCommandFunc(cmd *cobra.Command, args []string) {
 			// endpoint is health.
 			ctx, cancel := commandCtx(cmd)
 			_, err = cli.Get(ctx, "health")
-			cancel()
 			eh := epHealth{Ep: ep, Health: false, Took: time.Since(st).String()}
 			// permission denied is OK since proposal goes through consensus to get it
 			if err == nil || err == rpctypes.ErrPermissionDenied {
 				eh.Health = true
 			} else {
 				eh.Error = err.Error()
 			}
+
+			if eh.Health == true {
+				resp, err := cli.AlarmList(ctx)
+				if err == nil && len(resp.Alarms) > 0 {
```

Comment:
if `err != nil`, `eh.Health` should be `false`?

Reply:
@jingyih By default, eh.Health is set to false. If the "cli.get" call fails, it means key-based health check has failed and hence no need to check for alarm-based health check. Only if key-based health check succeeds (which means eh.Health is set to true), alarm-based health check is performed.

Reply:
Oh sorry I was referring to the `err` on line 138. If we cannot get alarm list, i.e., `cli.AlarmList` returns a non-nil err, the health check should also fail, right?

Reply:
@jingyih Oh sorry for missing this. Made code change and committed again.

Reply:
@jingyih Can you please review this and commit (if change is acceptable)?